docs/internal.yml

```diff
@@ -0,0 +1,245 @@
+        - Internal API
+      operationId: Provision device
+      summary: Register a new device with the deviceconfig service.
+      parameters:
+        - in: path
+          name: tenantId
+          schema:
+            type: string
+          required: true
+          description: ID of tenant the device belongs to.
+      requestBody:
+        content:
+          application/json:
+            schema:
+              $ref: '#/components/schemas/NewDevice'
+      responses:
+        204:
+          description: Device provisioned successfully.
+        400:
+          $ref: '#/components/responses/InvalidRequestError'
+        500:
+          $ref: '#/components/responses/InternalServerError'
+
+  /tenants/{tenantId}/devices/{deviceId}:
+    delete:
+      tags:
+        - Internal API
+      operationId: Decommission device
+      summary: Remove a device from Iot Hub.
+      parameters:
+        - in: path
+          name: tenantId
+          schema:
+            type: string
+          required: true
+          description: ID of tenant the device belongs to.
+        - in: path
+          name: deviceId
+          schema:
+            type: string
+          required: true
+          description: ID of the target device.
+      responses:
+        204:
+          description: Device deleted from Iot Hub
+        500:
+          $ref: '#/components/responses/InternalServerError'
+
+
+  /tenants/{tenantId}/bulk/devices/status:
+    put:
+      operationId: Update device statuses
+      tags:
+        - Internal API
+      summary: Update device statuses in bulk.
+      parameters:
+        - in: path
+          name: tenantId
+          schema:
+            type: string
+          required: true
+          description: ID of tenant the device belongs to.
+      requestBody:
+        content:
+          application/json:
+            schema:
+              type: object
+              properties:
+                device_ids:
+                  type: array
+                  items:
+                    type: string
+                  description: |
+                    List of device IDs to update.
+                    Up to 100 devices can be processed per request.
+                status:
+                  type: string
+                  enum:
+                    - enabled
+                    - disabled
+                  description: New status for list of devices.
+              required:
+                - device_ids
+                - status
+      responses:
+        200:
+          description: >-
+            Bulk request processed successfully. Check response body for individual
+            bulk item statuses.
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/BulkResult'
+        400:
+          description: Bad Request.
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/Error'
+        404:
```

Comment:
I think this endpoint will never return 404 or 500

Reply:
True. Or, technically it can still return 500 on Go panic, but never 404.